dynamic_programming/kadanes_algorithm.dart

```diff
@@ -0,0 +1,34 @@
+import 'dart:math';
+
+int kadanesAlgorithm(List<int> array) {
+  int maxEndingHere = array[0];
+  int maxSoFar = array[0];
+
+  for (int num in array.sublist(1, array.length)) {
+    maxEndingHere = [maxEndingHere + num, num].reduce(max);
+    maxSoFar = [maxSoFar, maxEndingHere].reduce(max);
```

Comment:
From what i see you could just 
```suggestion
    maxEndingHere = max(maxEndingHere + num, num);
    maxSoFar = max(maxSoFar, maxEndingHere);
```

Reply:
@Parowicz  thanks for the suggestion. I'll update with the suggested changes